cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunNativeImageTestTask.java

```diff
@@ -370,6 +382,35 @@ private int runTestSuiteWithNativeImage(Package currentPackage, JBallerinaBacken
         }
     }
 
+    private void validateResourcesWithinJar(Map<String, TestSuite> testSuiteMap, String packageName)
+                                            throws IOException {
+        TestSuite testSuite = testSuiteMap.values().toArray(new TestSuite[0])[0];
+        List<String> dependencies = testSuite.getTestExecutionDependencies();
+        String jarPath = "";
+        for (String dependency : dependencies) {
+            if (dependency.contains(packageName + ".jar")) {
+                jarPath = dependency;
+            }
+        }
+        try (ZipInputStream jarInputStream = new ZipInputStream(new FileInputStream(jarPath))) {
+            ZipEntry entry;
+            boolean isResourceExist = false;
+            while ((entry = jarInputStream.getNextEntry()) != null) {
+                String path = entry.getName();
+                if (path.startsWith("resources/")) {
+                    isResourceExist = true;
+                }
+                jarInputStream.closeEntry();
```

Comment:
This is redundant. Closing the stream is intrinsic to try-with-resource

Reply:
Here we are closing each file entry within ZipInputStream. However I should not use `jarInputStream.close()` here.